allen-cell-animated/simularium-website
# Feature/load file from URL

Problem
=======
Resolves #144 

Solution
========
* Changed the text/ordering of exiting items under the "Load model" menu per spec
    ![image](https://user-images.githubusercontent.com/12690133/131423769-5c38ccf8-e94f-4e86-9811-5c1c469556ed.png)

* Added a new "Load by URL" menu item that opens a modal when clicked.
    ![image](https://user-images.githubusercontent.com/12690133/131424200-a20624f1-d0fd-42e6-a78c-15369ad1e0f2.png)
    
    When user clicks "Load" without entering anything:
    ![image](https://user-images.githubusercontent.com/12690133/131424303-87234597-3f58-4c14-aef0-a468c695ac40.png)

    When user enters an invalid URL (this URL validation/notification is not new):
    ![image](https://user-images.githubusercontent.com/12690133/131424435-485e9813-c2fa-4df1-8c03-73f20dad4758.png)

* If the user inputs a valid URL to a simularium file then the modal closes, the URL changes to the `trajUrl` URL they can copy if they want, and the trajectory loads. 
* If the user clicks on Load model -> Load by URL while they are on the landing page or the Quick Start page (not on the viewer page), then they get redirected to the viewer page and then the modal opens. I made it this way so it's consistent with the existing behavior for local file uploads.

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. `npm start` and make sure everything works as described above
